flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/ZooKeeperStateHandle.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.zookeeper;
+
+import org.apache.flink.runtime.state.StateHandle;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A {@link StateHandle} with its path in ZooKeeper.
+ *
+ * @param <T> Type of state
+ */
+public class ZooKeeperStateHandle<T> {
+
+	private final StateHandle<T> stateHandle;
+
+	private final String pathInZooKeeper;
+
+	public ZooKeeperStateHandle(StateHandle<T> stateHandle, String pathInZooKeeper) {
+		this.stateHandle = checkNotNull(stateHandle, "State handle");
+		this.pathInZooKeeper = checkNotNull(pathInZooKeeper, "Path in ZooKeeper");
+	}
+
+	public StateHandle<T> getStateHandle() {
```

Comment:
This is confusing because it creates something like: `zookeeperStateHandle.getStateHandle().getState(...)`. Why calling `getStateHandle` on a `StateHandle`? The name indicates that `ZooKeeperStateHandle` is already a `StateHandle`. Why not simply offer a `getState(ClassLoader loader)` which forwards the call to the wrapped `StateHandle`?